### ISSUE TITLE
Error on attempt to use nonexistent component

### ISSUE THREAD

**Issue by RothAndrew** [defenseunicorns/zarf#92]
If I try to use `zarf init --components foo` or `zarf package deploy --components foo` but the component `foo` doesn't exist, Zarf should throw an error and refuse to continue.

My test has `zarf init --confirm --components management,logging,utility-cluster --host localhost`, but the `utility-cluster` component got renamed to `gitops-service`. Zarf happily continues, without deploying the gitops service because I didn't specify its name, but I meant to.

**Comment by jeff-mccoy** [comment]
Yeah that's not very intuitive.  That code is pretty dumb right now and needs some cleanup as it was more of an afterthought than well put together.
